merlin/study/script_adapter.py

```diff
@@ -35,13 +35,160 @@
+    """
+    A SchedulerScriptAdapter class for slurm blocking parallel launches,
+    the SlurmScriptAdapter uses non-blocking submits.
+    """
+
+    key = "merlin-slurm"
+
+    def __init__(self, **kwargs):
+        """
+        Initialize an instance of the MerinSlurmScriptAdapter.
+        The MerlinSlurmScriptAdapter is the adapter that is used for workflows that
+        will execute flux parallel jobs in a celery worker. The only configurable aspect to
+        this adapter is the shell that scripts are executed in.
+
+        :param **kwargs: A dictionary with default settings for the adapter.
+        """
+        super(MerlinSlurmScriptAdapter, self).__init__(**kwargs)
+
+        new_unsupported = [
+            "task_queue",
+            "max_retries",
+            "pre",
+            "post",
+            "gpus per task",
+            "walltime",
+            "gpus",
+            "restart",
+        ]
+        self._unsupported = set(list(self._unsupported) + new_unsupported)
+
+    def get_header(self, step):
+        """
+        Generate the header present at the top of Slurm execution scripts.
+
+        :param step: A StudyStep instance.
+        :returns: A string of the header based on internal batch parameters and
+            the parameter step.
+        """
+        return "#!{}".format(self._exec)
+
+
+class MerlinFluxScriptAdapter(MerlinSlurmScriptAdapter):
+    """
+    A SchedulerScriptAdapter class for flux blocking parallel launches,
+    the FluxScriptAdapter uses non-blocking submits.
+    """
+
+    key = "merlin-flux"
+
+    def __init__(self, **kwargs):
+        """
+        Initialize an instance of the MerinFluxScriptAdapter.
+        The MerlinFluxScriptAdapter is the adapter that is used for workflows that
+        will execute flux parallel jobs in a celery worker. The only configurable aspect to
+        this adapter is the shell that scripts are executed in.
+
+        :param **kwargs: A dictionary with default settings for the adapter.
+        """
+        flux_version = kwargs.pop("flux_version", "0.13.0")
+        super(MerlinFluxScriptAdapter, self).__init__(**kwargs)
+
+        #  "cmd": "flux mini run",
+        self._cmd_flags = {
+            "cmd": "flux mini run",
+            "ntasks": "-n",
+            "nodes": "-N",
+            "cores per task": "-c",
+            "gpus per task": "-g",
+        }
+
+        # Flux versions before 0.13 had a different run command
+        vers = [int(n) for n in flux_version.split(".")]
+        if vers[0] == 0 and vers[1] < 13:
+            self._cmd_flags["cmd"] = "flux wreckrun"
+
+        new_unsupported = [
+            "cmd",
+            "ntasks",
+            "nodes",
+            "gpus",
+            "walltime",
+            "reservation",
+            "restart",
+            "task_queue",
+            "max_retries",
+            "pre",
+            "post",
+            "depends",
+        ]
+        self._unsupported = set(new_unsupported)
```

Comment:
Isn't this equivalent to just using curly brackets instead? `new_unsupported = {...`

Reply:
That file was added before the move, this PR is from Nov/19.

Reply:
The previous comment was for the flux_local.yaml file.

Reply:
Yes, it was left over from having two lists , then making a set.